src/commonMain/kotlin/fr/acinq/lightning/crypto/LocalKeyManager.kt

```diff
@@ -3,168 +3,166 @@ package fr.acinq.lightning.crypto
+         *
+         * @param fundingPubKey funding public key
+         * @return a BIP32 path
+         */
+        fun channelKeyPath(fundingPubKey: PublicKey): KeyPath {
+            val buffer = fundingPubKey.value.sha256().toByteArray()
+
+            val path = sequence {
+                var i = 0
+                while (true) {
+                    yield(Pack.int32BE(buffer, i).toUInt().toLong())
+                    i += 4
+                }
+            }
+            return KeyPath(path.take(8).toList())
         }
 
-        /** Path for node keys generated by eclair-core */
-        fun eclairNodeKeyBasePath(chainHash: ByteVector32) = when (chainHash) {
-            Block.RegtestGenesisBlock.hash, Block.TestnetGenesisBlock.hash -> listOf(hardened(46), hardened(0))
-            Block.LivenetGenesisBlock.hash -> listOf(hardened(47), hardened(0))
-            else -> throw IllegalArgumentException("unknown chain hash $chainHash")
+        fun channelKeyBasePath(chain: Chain) = when (chain) {
+            Chain.Regtest, Chain.Testnet -> KeyPath.empty I hardened(48) I hardened(1)
+            Chain.Mainnet -> KeyPath.empty I hardened(50) I hardened(1)
         }
 
-        // WARNING: if you change this path, you will change your node id even if the seed remains the same!!!
-        // Note that the node path and the above channel path are on different branches so even if the
-        // node key is compromised there is no way to retrieve the wallet keys
-        fun nodeKeyBasePath(chainHash: ByteVector32) = when (chainHash) {
-            Block.RegtestGenesisBlock.hash, Block.TestnetGenesisBlock.hash -> listOf(hardened(48), hardened(0))
-            Block.LivenetGenesisBlock.hash -> listOf(hardened(50), hardened(0))
-            else -> throw IllegalArgumentException("unknown chain hash $chainHash")
+        /** Path for node keys generated by eclair-core */
+        @Deprecated("used for backward-compat with eclair-core", replaceWith = ReplaceWith("nodeKeyBasePath(chain)"))
+        fun eclairNodeKeyBasePath(chain: Chain) = when (chain) {
+            Chain.Regtest, Chain.Testnet -> KeyPath.empty I hardened(46) I hardened(0)
+            Chain.Mainnet -> KeyPath.empty I hardened(47) I hardened(0)
         }
 
-        fun bip84BasePath(chainHash: ByteVector32) = when (chainHash) {
-            Block.TestnetGenesisBlock.hash, Block.RegtestGenesisBlock.hash -> listOf(hardened(84), hardened(1))
-            Block.LivenetGenesisBlock.hash -> listOf(hardened(84), hardened(0))
-            else -> throw IllegalArgumentException("unknown chain hash $chainHash")
+        fun nodeKeyBasePath(chain: Chain) = when (chain) {
+            Chain.Regtest, Chain.Testnet -> KeyPath.empty I hardened(48) I hardened(0)
+            Chain.Mainnet -> KeyPath.empty I hardened(50) I hardened(0)
         }
     }
 }
+
+infix fun KeyPath.I(index: Long): KeyPath = this.append(index)
+
+infix fun KeyPath.I(other: KeyPath): KeyPath = this.append(other)
```

Comment:
This is may not 100% orthodox, but `/` is forbiden as a function name, and it renders nicely in Intellij and makes the paths human-readable.

Reply:
Actually you can override `/`, which is a bit more intuitive:

```suggestion
infix operator fun KeyPath.div(index: Long): KeyPath = this.append(index)

infix operator fun KeyPath.div(other: KeyPath): KeyPath = this.append(other)
```

Reply:
Done in 92eadb53.